### DISCUSSION TITLE
Spring Boot + Next.js 端口唯一代码分离方案 | 夜天之书

### DISCUSSION THREAD

**Discussion by giscus** [tisonkun/blog#48] [Comments]
# Spring Boot + Next.js 端口唯一代码分离方案 | 夜天之书

上周开发报表的时候实现了一个 Spring Boot + Next.js 前后端代码分离，但是端口唯一的 WebApp 方案。这个方案既能利用好各自的生态，也能避免要起两个 server 占用两个端口的繁琐构建部署逻辑。 本文从选型到实现做一个分享，主要面向熟悉 Spring Boot 技术栈，又想跟进 React 系前端组件化方案的开发者。

https://www.tisonkun.org/2022/10/22/springboot-nextjs/

**Comment by TomorJM** [comment]
这个很赞

**Comment by tageecc** [comment]
感觉这个技术方案是强行nextjs，ssr基本没发挥价值，最多有csr的效果，还不如走完整的前后端分离方案，随便搭个webpack或者vite仓库
nextjs可以作为服务前端应用，springbot通过rpc调用方式去协作会更合理
